src/main/java/org/broadinstitute/hdf5/HDF5File.java

```diff
@@ -63,7 +65,15 @@
      * @param file the underlying file name.
      */
     public HDF5File(final File file) {
-        this(file, OpenMode.READ_ONLY);
+        this(file.toPath());
```

Comment:
This should have a check for null here to avoid NPR.  ex

` this(file == null ? null : file.toPath());`